src/processes.c

```diff
@@ -2590,37 +2461,33 @@ static int ps_read (void)
 					break;
 			}
 
-			pse.cpu_user = 0;
 			/* tv_usec is nanosec ??? */
-			pse.cpu_user_counter = procentry[i].pi_ru.ru_utime.tv_sec * 1000000 +
+			pse.counters.cpu_user = procentry[i].pi_ru.ru_utime.tv_sec * 1000000 +
 				procentry[i].pi_ru.ru_utime.tv_usec / 1000;
 
-			pse.cpu_system = 0;
 			/* tv_usec is nanosec ??? */
-			pse.cpu_system_counter = procentry[i].pi_ru.ru_stime.tv_sec * 1000000 +
+			pse.counters.cpu_system = procentry[i].pi_ru.ru_stime.tv_sec * 1000000 +
 				procentry[i].pi_ru.ru_stime.tv_usec / 1000;
 
-			pse.vmem_minflt = 0;
-			pse.vmem_minflt_counter = procentry[i].pi_minflt;
-			pse.vmem_majflt = 0;
-			pse.vmem_majflt_counter = procentry[i].pi_majflt;
+			pse.counters.vmem_minflt = procentry[i].pi_minflt;
+			pse.counters.vmem_majflt = procentry[i].pi_majflt;
 
-			pse.vmem_size = procentry[i].pi_tsize + procentry[i].pi_dvm * pagesize;
-			pse.vmem_rss = (procentry[i].pi_drss + procentry[i].pi_trss) * pagesize;
+			pse.gauges.vmem_size = procentry[i].pi_tsize + procentry[i].pi_dvm * pagesize;
+			pse.gauges.vmem_rss = (procentry[i].pi_drss + procentry[i].pi_trss) * pagesize;
 			/* Not supported */
-			pse.vmem_data = 0;
-			pse.vmem_code = 0;
-			pse.stack_size =  0;
+			pse.gauges.vmem_data = 0;
+			pse.gauges.vmem_code = 0;
+			pse.gauges.stack_size =  0;
 
-			pse.io_rchar = -1;
-			pse.io_wchar = -1;
-			pse.io_syscr = -1;
-			pse.io_syscw = -1;
-			pse.io_diskr = -1;
-			pse.io_diskw = -1;
+			pse.gauges.io_rchar = -1;
```

Comment:
I wonder if this can be accomplished by simply assigning `procstat_gauges_init` to `pse.gauges` before reading the values... In fact, can we do this right after we declare `pse`?